script/csharp/compile-command.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+# This script does two things
+#   1. Redirects all output to stderr
+#   2. Supplies .dll references to the csc command
+
+# Actually compile:
+# [dotnet] [csc] -reference:[stdlib.dll] [...passed in args...]
+VERSION=$(dotnet --version)
+/usr/bin/dotnet "/usr/share/dotnet/sdk/"$VERSION"/Roslyn/bincore/csc.dll" \
+-reference:"/usr/share/dotnet/sdk/"$VERSION"/ref/netstandard.dll" "$@" 1>&2
```

Comment:
> No need for full path to dotnet

Why then have we been specifying `/usr/bin/g++` etc for the rest of them? It would be more readable to keep it short. I tested this with g++ and it worked but python did not work without `/usr/bin`.

> csc.dll prints version number of the C# compiler, I think the .NET Core SDK version should be added

Makes sense. Do we bother removing the copyright line in keeping with GHC and GCC?

Reply:
> Why then have we been specifying `/usr/bin/g++` etc for the rest of them? It would be more readable to keep it short. I tested this with g++ and it worked but python did not work without `/usr/bin`. Perhaps further investigation in another PR.

When a command (array of arguments) is executed using isolate, the first argument needs to be the full path to the program and the remaining arguments are passed as argv[1] etc. At first, the compilation commands were executed directly using isolate so the first argument needed to be the full path to the compiler (e.g. `/usr/bin/g++`).

When Java was added (32dcf29af8ed249792e73d744af97f29d16d8b00) it required two commands during compilation, javac and jar, so the compilation commands were wrapped in bash (e.g. first arg = `/bin/bash`, second arg = `-c`, third arg = `javac ... && jar ...`). Since the compilation command is now executed by bash, it can use shell operators like &&. It also doesn't need to use full paths to programs because bash searches for them in $PATH.

Python still requires a full path is because it uses an interpreter, and the interpreter commands aren't wrapped in bash.

All this is only for commands in db/languages.yml that are going to be executed directly by isolate, i.e. interpreter commands and (historically) compilation commands. Inside a bash script like `compile-command.sh` there is no need to use full paths to programs because bash searches in $PATH.

> > csc.dll prints version number of the C# compiler, I think the .NET Core SDK version should be added
> 
> Makes sense. Do we bother removing the copyright line in keeping with GHC and GCC?

There is an option `-nologo` that removes the copyright message, so sure.

Reply:
Ah okay. Makes sense. In which case, I'd like to propose that we remove `/usr/bin` from all the current compile commands to make them more readable to users. I'll open another PR